.devcontainer/advanced-integration/devcontainer.json

```diff
@@ -0,0 +1,39 @@
+// For more details, see https://aka.ms/devcontainer.json.
+{
+	"name": "Advanced Integration",
+	"image": "mcr.microsoft.com/devcontainers/javascript-node:0-20",
+	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}/advanced-integration",
+
+	// Use 'onCreateCommand' to run commands when creating the container
+	"onCreateCommand": "if [ -n \"$CODESPACES\" ]; then sudo cp ../.devcontainer/advanced-integration/welcome-message.txt /usr/local/etc/vscode-dev-containers/first-run-notice.txt; fi;",
+
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "npm install",
+
+	// Use 'postAttachCommand' to run commands when attaching to the container.
+	"postAttachCommand": {
+		"Start server": "npm start"
+	},
+
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [
+		8888
+	],
+	"portsAttributes": {
+		"8888": {
```

Comment:
Can we set a port attribute to make this open in the simple browser inside VS Code instead of asking the user to open it in the browser themselves? If you make this change, you could also change the welcome message and readme to match.

Reply:
This simplified the flow very nicely, however, the sample has few `alert()`s which shows a modal. However, in the VS Code browser, I get 👇 

![image](https://user-images.githubusercontent.com/24955913/234719804-b195e22b-b996-44b0-a2ca-8dbac48cfe47.png)

From https://github.com/Microsoft/vscode/issues/67109 doesn't look like there's a way to show the modal as it's by design. We could switch to VS Code APIs but I don't think we should change the sample code 😕 

@jkeech not sure if you know of any other workaround.

Reply:
In that case, probably switch to `openBrowser` or `openBrowserOnce`: https://github.com/devcontainers/spec/blob/main/schemas/devContainer.base.schema.json#L72-L81